lib/foreman_maintain/executable.rb

```diff
@@ -40,5 +43,17 @@ def __run__(execution)
       @_execution = execution
       run
     end
+
+    # method defined both on object and class to ensure we work always with object
+    # even when the definitions provide us only class
+    def ensure_instance
+      self
+    end
+
+    class << self
+      def ensure_instance
```

Comment:
While I was looking at removing this, I started hitting more issues, as we return classes some places (such as `find_checks`), and we need to turn them into instances at some point. I'm for keeping the current model for now, and instanciating the classes if needed (+ use the `manual_detection` metadata for classes that can't be initialized automatically)